### PR TITLE
Removed file path from warning messages

### LINKED ISSUE #619: Remove file paths from diktat warning messages
In some warnings we put file path into `freeText`, but ktlint already shows coordinates in a file, and obviously file name too, so this is redundant.
The only thing that might require consideration: when file path is analyzed when choosing correct package name.

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Changed code
* Changed tests

This pull request closes #619

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt [hunk 1]
```diff
@@ -88,7 +83,8 @@ class HeaderCommentRule(private val configRules: List<RulesConfig>) : Rule("head
                 val numDeclaredClassesAndObjects = node.getAllChildrenWithType(ElementType.CLASS).size +
                         node.getAllChildrenWithType(ElementType.OBJECT_DECLARATION).size
                 if (numDeclaredClassesAndObjects == 0 || numDeclaredClassesAndObjects > 1) {
```
Comment (kentr0w) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt:85]:
```suggestion
   if (numDeclaredClassesAndObjects != 1) {
```

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt [hunk 2]
```diff
@@ -162,7 +158,13 @@ class HeaderCommentRule(private val configRules: List<RulesConfig>) : Rule("head
                 copyrightWords.any { commentNode.text.contains(it, ignoreCase = true) }
             }
         if (isWrongCopyright || isMissingCopyright || isCopyrightInsideKdoc) {
-            HEADER_MISSING_OR_WRONG_COPYRIGHT.warnAndFix(configRules, emitWarn, isFixMode, fileName, node.startOffset, node) {
+            val freeText = when {
```
Comment (kentr0w) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt:161]: What if several flags are true? Maybe better to do string concatenation?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileSize.kt [hunk 3]
```diff
@@ -27,31 +31,18 @@ class FileSize(private val configRules: List<RulesConfig>) : Rule("file-size") {
...
-
-            val realFilePath = calculateFilePath(fileName)
-
-            if (!realFilePath.contains(SRC_PATH)) {
+            val filePathParts = node.getFilePath().splitPathToDirs()
+            if (SRC_PATH !in filePathParts) {
                 log.error("$SRC_PATH directory is not found in file path")
             } else {
-                if (ignoreFolders.none { realFilePath.containsAll(it.splitPathToDirs()) }) {
+                val ignoreFolders = configuration.ignoreFolders
```
Comment (kentr0w) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileSize.kt:38]: Maybe it's better here without a variable?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/IndentationRule.kt [hunk 4]
```diff
@@ -132,6 +133,7 @@ class IndentationRule(private val configRules: List<RulesConfig>) : Rule("indent
             val numBlankLinesAfter = lastChild.text.count { it == '\n' }
             if (lastChild.elementType != WHITE_SPACE || numBlankLinesAfter != 1) {
                 val warnText = if (lastChild.elementType != WHITE_SPACE || numBlankLinesAfter == 0) "no newline" else "too many blank lines"
+                val fileName = filePath.substringAfterLast(File.separator)
```
Comment (kentr0w) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/IndentationRule.kt:136]: The variable `filePath` is used only here. Maybe it's better here without a variable too?

**Review by kentr0w — APPROVED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt [hunk 5]
```diff
@@ -105,7 +101,7 @@ class HeaderCommentRule(private val configRules: List<RulesConfig>) : Rule("head
         val firstKdoc = node.findChildAfter(IMPORT_LIST, KDOC)
         // if `firstKdoc.treeParent` is File then it's a KDoc not bound to any other structures
         if (node.findChildBefore(PACKAGE_DIRECTIVE, KDOC) == null && firstKdoc != null && firstKdoc.treeParent.elementType == FILE) {
-            HEADER_NOT_BEFORE_PACKAGE.warnAndFix(configRules, emitWarn, isFixMode, fileName, firstKdoc.startOffset, firstKdoc) {
+            HEADER_NOT_BEFORE_PACKAGE.warnAndFix(configRules, emitWarn, isFixMode, "header KDoc is located after imports", firstKdoc.startOffset, firstKdoc) {
```
Comment (aktsay6) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt:104]: May be it's better to say "header KDoc is located after imports or package" based on rule name?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/FileNaming.kt [hunk 6]
```diff
@@ -38,24 +38,24 @@ class FileNaming(private val configRules: List<RulesConfig>) : Rule("file-naming
...
+        val (name, extension) = getFileParts(filePath)
         if (!name.isPascalCase() || !validExtensions.contains(extension)) {
             FILE_NAME_INCORRECT.warnAndFix(configRules, emitWarn, isFixMode, "$name$extension", 0, node) {
                 // FixMe: we can add an autocorrect here in future, but is there any purpose to change file or class name?
             }
         }
     }
 
-    @Suppress("UnsafeCallOnNullableType")
+    @Suppress("UnsafeCallOnNullableType", "ControlFlowWithEmptyBody")
```
[the remark this answers is not in the document]
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/FileNaming.kt:56]: Kotlin compiler warns that there is an empty `else` branch, but there is a fixme comment so it's kind of expected.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileSize.kt (hunk 3 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileSize.kt:38]: Yes, makes sense

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/IndentationRule.kt (hunk 4 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/IndentationRule.kt:136]: Here it helps not to overcomplicate the string template, so I think it's fine to leave a variable

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt (hunk 5 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt:104]: Yes, fixed

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt (hunk 2 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt:161]: I think it doesn't affect meaning:
* `isMissingCopyright` means block comment doesn't exist
* `isWrongCopyright` means it exists but doesn't have keywords
* `isCopyrightInsideKdoc` means block comment doesn't (which makes `isMissingCopyright` `true`), but there is copyright information inside KDoc, which makes description of `isMissingCopyright` redundant here

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt:161]: But you are right, it still needs some refinement

**Pull request merged**